src/container/srv_target.c

```diff
@@ -438,25 +458,41 @@ ds_cont_cache_fini(void)
 static int
 cont_child_destroy_one(void *vin)
 {
-	struct cont_tgt_destroy_in     *in = vin;
 	struct dsm_tls		       *tls = dsm_tls_get();
+	struct cont_tgt_destroy_in     *in = vin;
 	struct ds_pool_child	       *pool;
-	struct ds_cont_child	       *cont;
 	int				rc;
 
 	pool = ds_pool_child_lookup(in->tdi_pool_uuid);
 	if (pool == NULL)
 		D_GOTO(out, rc = -DER_NO_HDL);
 
-	rc = cont_child_lookup(tls->dt_cont_cache, in->tdi_uuid, NULL,
-			       &cont);
-	if (rc == 0) {
+	while (1) {
+		struct ds_cont_child *cont;
+		bool		      resyncing = false;
+
+		rc = cont_child_lookup(tls->dt_cont_cache, in->tdi_uuid, NULL,
+				       &cont);
+		if (rc == -DER_NONEXIST)
+			break;
+		if (rc != 0)
+			D_GOTO(out_pool, rc);
+		/* found it */
+
+		ABT_mutex_lock(cont->sc_mutex);
+		cont->sc_destroying = 1;
+		if (cont->sc_dtx_resyncing) {
+			resyncing = true;
+			ABT_cond_wait(cont->sc_dtx_resync_cond, cont->sc_mutex);
```

Comment:
You hold the sc_mutex to wait the ABT conditions, that will block the dtx_resync() to reset the sc_dtx_resyncing at line 328.

Reply:
A calls ABT_cond_wait, then A will be release for the mutex automatically and wait on the cond, then
B can do the resync, hold the mutex and clear the flag and set the cond;
A wakeup and automatically hold the mutex again and return from ABT_cond_wait.
it seems fine?

Reply:
Thanks for the explanation. It's fine.